tests/functional_small/solvers_nonlinear/newton_raphson/eigen_w_basic_backtrack.cc

```diff
@@ -0,0 +1,70 @@
+//
+#include "pressio/solvers.hpp"
+struct ValidSystem 
+{
+  using state_type = Eigen::VectorXd;
+  using residual_type = state_type;
+  using jacobian_type = Eigen::SparseMatrix<double>;
+
+  state_type createState() const {
+    return state_type(2);
+  }
+
+  residual_type createResidual() const {
+    return residual_type(2);
+  }
+
+  jacobian_type createJacobian() const {
+    return jacobian_type(2, 2);
+  }
+
+  void residual(const state_type& x,
+                residual_type& res) const
+  {
+    res(0) =  x(0)*x(0) + x(1)*x(1) - 4.0;
+    res(1) = x(0)*x(1)  - 1.0;
+  }
+
+  void jacobian(const state_type& x, jacobian_type& jac) const {
+    jac.coeffRef(0, 0) = 2*x(0);
+    jac.coeffRef(0, 1) = 2*x(1);
+    // Have incorrect entries to mimic requirement for line search
```

Comment:
why does "to mimic requirements for line search" mean?

Reply:
Changed to "Have incorrect entries to require line search"